packages/aggregator/src/analytics/aggregateAnalytics/aggregations/getFinalValuePerPeriod.js

```diff
@@ -106,12 +106,19 @@ class FinalValueAggregator {
     this.cache = cache;
   }
 
-  getContinuousValues(analytics, aggregationPeriod) {
-    const periods = getContinuousPeriodsForAnalytics(analytics, aggregationPeriod, true);
+  getFilledValues(analytics, aggregationPeriod, fillEmptyPeriodsWith) {
+    const checkIfFillingContinuousValues = fillEmptyPeriodsWith === 'previous';
+    const periods = getContinuousPeriodsForAnalytics(
+      analytics,
+      aggregationPeriod,
+      checkIfFillingContinuousValues, // To get a continuous period till now, we need to set true for this option config (continueTilCurrentPeriod). This can be a new option config in the future.
+    );
 
     const values = [];
     this.cache.iterateOrganisationUnitCache(organisationUnitCache => {
-      let mostRecentValue;
+      let mostRecentValue = checkIfFillingContinuousValues
+        ? undefined
+        : Object.values(organisationUnitCache)[0];
```

Comment:
Here we want to fill empty period with value 'test' in two orgUnits, see this test case as example. Screenshot: 
<img width="797" alt="11620283114_ pic_hd copy" src="https://user-images.githubusercontent.com/31789355/117252879-1644b280-ae36-11eb-8e7a-985191648658.png">

In this case `period` is '20210101 - 20210107'. We need a preset `mostRecentValue`, if not, it will start filling from `20210104` where org2 has value.